src/main/java/com/mihir/pipposample/Employee/EmployeeService.java

```diff
@@ -0,0 +1,75 @@
+package com.mihir.pipposample.Employee;
+
+import com.google.inject.Inject;
+import com.mihir.pipposample.Employee.EmployeeDao;
+import org.bson.Document;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class EmployeeService {
+    @Inject
+    EmployeeDao employeeDao;
+
+    public void createEmployee(Document employee) {
+        try{
```

Comment:
your controller should not have context of bson doc
create a pojo ->bsondoc
bson doc should only be an argument in mongodao